content/web-api-rest/reading/rest-fundamentals/_index.md

```diff
@@ -0,0 +1,444 @@
+{
+    "Title": "Foo"
+}
+```
+
+`Foo` does not contain enough characters to be a valid CodingEvent title. The CodingEvents API response to such a request therefore includes a `400` status code. 
+This alerts the client that they must correct their data representation. The response body indicates which aspects of the request are invalid. This is a 400 failed response body:
+
+```bash
+{
+    "error": "invalid fields",
+    "fields": [
+    {
+        "Title": "must be between 10 and 100 characters in length"
+    }
+    ]
+}
+```
+
+Using the information in the response, the client can fix their request body and reissue the request successfully.
+{{% /notice %}}
+
+{{% notice blue "Note" "rocket" %}}
+The `401`, or `Unauthorized`, status code actually indicates that the consumer is not authenticated. This means the consumer has not proven their identity to the API. The `403`, or `Forbidden`, status code is a more accurate description of being unauthorized. After authenticating, the consumer's authorization can determine if they are allowed or forbidden from taking the requested action.
+{{% /notice %}}
+
+#### Headers
+
+In RESTful design, HTTP headers communicate metadata about each interaction with a resource.
+
+**Common request/response headers in REST:**
+
+| Request/Response | Header | Meaning | Example |
+|-----|-----|-----|-----|
+| Both | `Content-Type` | The attached body has the following media type | `application/json` |
+| Request | `Accept` | The client expects the requested resource representation in the given media type | `application/json` |
+| Response | `Location` | The created resource representation can be found at the given URL value | `/resources/{id}` |
+
+{{% notice green "Tip" "rocket" %}}
+The `Authorization` request header is also commonly used. 
+{{% /notice %}}
+
+### Learning More
+
+These articles have covered the fundamental aspects of the RESTful mental model and practical usage. However, RESTful design is a deep topic that even 
+extends beyond the web and use of HTTP! 
+
+If you want to learn more, the following links are a good start:
+
+#### Practical Understanding
+
+- [Craig Dennis: APIs for beginners (YouTube)](https://www.youtube.com/watch?v=GZvSYJDk-us&t=0s)
+- [REST sub-collections, relationships and links](https://restful-api-design.readthedocs.io/en/latest/relationships.html)
+- [OpenAPI specification & Swagger REST tools](https://swagger.io/specification/)
+- [The GitHub API](https://developer.github.com/v3/) and [Stripe (payment processing) API](https://stripe.com/docs/api) are excellent examples of RESTful design (and fantastic documentation)
+
+#### Deep Understanding
+
+- [the REST constraints](https://www.restapitutorial.com/lessons/whatisrest.html)
+- [The Richardson REST maturity model](https://www.martinfowler.com/articles/richardsonMaturityModel.html)
+- [the original REST doctoral thesis by Roy Fielding](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm)
+
```

Comment:
Please move this section to a Next Steps page. Delete Learning More subheading and rework the sentence at line 393 -> something like ->

We have covered the fundamental aspects of the RESTful mental model and practical usage. However, RESTful design is a deep topic that even 
extends beyond the web and use of HTTP!

Reply:
Implemented the above changes.